tests/components/holiday/test_calendar.py

```diff
@@ -0,0 +1,137 @@
+"""Tests for calendar platform of Holiday integration."""
+from datetime import datetime
+
+from freezegun.api import FrozenDateTimeFactory
+
+from homeassistant.components.calendar import (
+    DOMAIN as CALENDAR_DOMAIN,
+    SERVICE_GET_EVENTS,
+)
+from homeassistant.components.holiday.const import CONF_PROVINCE, DOMAIN
+from homeassistant.const import CONF_COUNTRY
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.setup import async_setup_component
+from homeassistant.util import dt as dt_util
+
+from tests.common import MockConfigEntry
+
+
+async def test_holiday_calendar_entity(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test HolidayCalendarEntity functionality."""
+    freezer.move_to(datetime(2023, 1, 1, 12, tzinfo=dt_util.UTC))
+
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_COUNTRY: "US", CONF_PROVINCE: "AK"},
+        title="United States, AK",
+    )
+    config_entry.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    await async_setup_component(hass, "calendar", {"calendar": {"platform": "demo"}})
+    await hass.async_block_till_done()
```

Comment:
```suggestion
```
We don't need `demo` for anything right?

Reply:
Right, was copy and paste…